rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,424 @@
+    
+    
+    molNameList = list(set(globals()['rdkit_mol_select_'+uid]))
+    globals()['selected_mols_view_'+uid].value = ', '.join(molNameList)
+    
+    
+    for i in molNameList:
+        mol = globals()['moldict_'+uid][i]
+        
+        if mol.GetNumConformers()>1:
+            
+            allConfIds = range(mol.GetNumConformers()-1)
+            globals()['confId_'+uid].options = allConfIds
+            
+            
+            if globals()['selectAllConfs_'+uid].value is True:
+                for i in allConfIds:
+                    globals()['rdkit_conf_select_'+uid].append(i)
+            else:
+                confId=globals()['confId_'+uid].value
+                if globals()['selectMultiConfs_'+uid].value is True:
+                    globals()['rdkit_conf_select_'+uid].append(confId)
+                else:
+                    globals()['rdkit_conf_select_'+uid] = list()
+                    globals()['rdkit_conf_select_'+uid].append(confId)
+                    
+            
+        elif mol.GetNumConformers()==1:
+            globals()['confId_'+uid].options=[0]
+            confId=globals()['confId_'+uid].value
+            globals()['rdkit_conf_select_'+uid]=list()
+            globals()['rdkit_conf_select_'+uid].append(confId)
+            globals()['selected_confs_view_'+uid].value = str(confId)
+        
+        
+    confIdsList = list(set(globals()['rdkit_conf_select_'+uid]))
+    globals()['selected_confs_view_'+uid].value = ', '.join([str(x) for x in confIdsList])
+    
+    
+    # Add molecule to viewer
+    for molName in molNameList:
+        for confId in confIdsList:
+            mol = globals()['moldict_'+uid][molName]
+            mb = Chem.MolToMolBlock(mol,confId=confId)
+            globals()['view_'+uid].addModel(mb,'sdf')
+    
+    
+    if len(molNameList)==1:
+        mol = globals()['moldict_'+uid][molDictKey]
+        # For precalculated property
+        try:
+            all_prop_from_mol=list(mol.GetPropNames())
+            if len(all_prop_from_mol)>0:
+                # update widget
+                globals()['prop_precalc_wg_'+uid].options=all_prop_from_mol
+                # Get selected property
+                prop_name=eval('prop_precalc_wg_'+uid+'.value')
+                prop_value=mol.GetProp(prop_name)
+                prop_str = prop_name + ' : ' + str(prop_value)
+                # Update viewer
+                globals()['prop_precalc_view_'+uid].value = prop_str
+            else:
+                globals()['prop_precalc_view_'+uid].value = 'No precalculated property found!'
+        except:
+            pass
+        
+        # For calculating rdkit supported property
+        try:
+            # Descriptor calculation schema eval("Descriptors.TPSA(mol)")
+            prop_name=globals()['prop_calc_wg_'+uid].value
+            prop_calc_cmd="Descriptors."+ prop_name + "(mol)"
+            prop_str = prop_name + ' : ' + str(eval(prop_calc_cmd))
```

Comment:
This is in fact great coding style in my sense. Thank you very much for teaching me that. This has been implemented too.